tests/integration/src/test/java/org/apache/pulsar/tests/integration/cluster/Cluster3Bookie2Broker.java

```diff
@@ -0,0 +1,317 @@
+                }
+            })
+        ;
+
+        broker2Container = new GenericContainer(NAME + "/" + IMG + ":" + "latest")
+            .withNetwork(network)
+            .withNetworkAliases(PULSAR_BROKER2_NAME)
+            .withLabel("cluster", "test")
+            .withLabel("service", "pulsar-broker")
+            .withExposedPorts(PULSAR_BROKER_PORT)
+            .withEnv("clusterName", "test")
+            .withEnv("zookeeperServers", "zookeeper")
+            .withEnv("configurationStore", "configuration-store:2184")
+            .withEnv("NO_AUTOSTART", "true")
+            .withCommand("sh", "-c", "bin/run-broker.sh")
+            .waitingFor(new LogMessageWaitStrategy()
+                .withRegEx(".*supervisord started with pid.*\\s")
+                .withTimes(1)
+                .withStartupTimeout(Duration.of(100, SECONDS)))
+            .withLogConsumer(new Slf4jLogConsumer(LOG).withPrefix(PULSAR_BROKER2_NAME))
+            .withCreateContainerCmdModifier(new Consumer<CreateContainerCmd>() {
+                @Override
+                public void accept(CreateContainerCmd createContainerCmd) {
+                    createContainerCmd.withHostName(PULSAR_BROKER2_NAME);
+                    createContainerCmd.withName(PULSAR_BROKER2_NAME + "-" + testName);
+                }
+            })
+        ;
+
+        proxyContainer = new GenericContainer(NAME + "/" + IMG + ":" + "latest")
+            .withNetwork(network)
+            .withNetworkAliases(PULSAR_PROXY_NAME)
+            .withLabel("cluster", "test")
+            .withLabel("service", "pulsar-proxy")
+            .withExposedPorts(PULSAR_BROKER_PORT)
+            .withEnv("clusterName", "test")
+            .withEnv("zookeeperServers", "zookeeper")
+            .withEnv("configurationStoreServers", "configuration-store:2184")
+            .withEnv("NO_AUTOSTART", "configuration-store:2184")
+            .withCommand("sh", "-c", "bin/run-proxy.sh")
+            .waitingFor(new LogMessageWaitStrategy()
+                .withRegEx(".*supervisord started with pid.*\\s")
+                .withTimes(1)
+                .withStartupTimeout(Duration.of(100, SECONDS)))
+            .withLogConsumer(new Slf4jLogConsumer(LOG).withPrefix(PULSAR_PROXY_NAME))
+            .withCreateContainerCmdModifier(new Consumer<CreateContainerCmd>() {
+                @Override
+                public void accept(CreateContainerCmd createContainerCmd) {
+                    createContainerCmd.withHostName(PULSAR_PROXY_NAME);
+                    createContainerCmd.withName(PULSAR_PROXY_NAME + "-" + testName);
+                }
+            })
+        ;
+
+        dockerClient = DockerClientFactory.instance().client();
+    }
+
+    public void execInBroker(String... commands) throws IOException, InterruptedException {
+        LOG.info("Executing Command in Broker : " + String.join(" ", commands));
+        Container.ExecResult execResult = broker1Container.execInContainer(commands);
```

Comment:
Is there a way to get the log the standard out before the command has completed? If something hangs while running, there'll be no way to tell what happened.

Reply:
I don't know , container streams are being tailed but I don't know if the exec command can be tailed.

Reply:
It can be done with docker-java which this is using test containers is using internally. You said on dev@ that you can access the docker-java client?

Reply:
yes docker client is available in the cluster .  Object will have to see what method can be used

Reply:
Since you have the docker-java client, you can use a lot of the stuff from PulsarClusterUtils and DockerUtils.